src/Generator/FileGenerator.php

```diff
@@ -504,10 +515,7 @@ public function generate()
         //build uses array
         foreach ($classes as $class) {
             //check for duplicate use statements
-            $uses = $class->getUses();
-            if (! empty($uses) && is_array($uses)) {
-                $classUses = array_merge($classUses, $uses);
-            }
+            $classUses = array_merge($classUses, $class->getUses());
```

Comment:
Do you want to use `array_unique` after the loop, as this won't do anything about duplicates.

Also, could this loop be moved inside the `if (! empty($uses)) {` conditional block as I don't see `$classUses` actually used anywhere else? I know this is beyond your change, but thought it worth mentioning.

Reply:
Overall OK, but that would change the current behavior, which I didn't touch, even if buggy.

I'd say that we should raise a separate issue about this, if relevant.

From my PoV, uses and aliases should disappear from `laminas/laminas-code`: they are confusing and unnecessary, in generated code.

Reply:
Practically: not touching this in this specific patch :)